features/cellular/framework/AT/AT_CellularNetwork.cpp

```diff
@@ -145,9 +145,25 @@ void AT_CellularNetwork::read_reg_params_and_compare(RegistrationType type)
             _connection_status_cb((nsapi_event_t)CellularRadioAccessTechnologyChanged, (intptr_t)&data);
         }
         if (reg_params._status != _reg_params._status) {
+            nsapi_event_t event_for_callback = (nsapi_event_t)CellularRegistrationStatusChanged;
+            intptr_t data_for_callback = (intptr_t)&data;
+            if (reg_params._status == StatusNotAvailable ||
+                    reg_params._status == NotRegistered ||
+                    reg_params._status == SearchingNetwork ||
+                    reg_params._status == RegistrationDenied ||
+                    reg_params._status == Unknown) {
+                if (!(_reg_params._status == StatusNotAvailable ||
```

Comment:
Hmm, if we drop to AttachedEmergencyOnly, RegisteredCSFBNotPreferredHome or to  RegisteredCSFBNotPreferredRoaming are we still registered? if we drop to RegisteredSMSOnlyHome or RegisteredSMSOnlyRoaming we are registered but we probably can't send data. @AriParkkila @mirelachirica what do you think? I guess there is no right answer here.

Reply:
Maybe it is ok if any other state than RegisteredHomeNetwork or RegisteredRoaming indicates NSAPI_STATUS_DISCONNECTED, since data connection is impossible in those states.